spec/controllers/session_controller_spec.rb

```diff
@@ -659,6 +659,22 @@ def post_login
         end
       end
     end
+
+    context 'rate limited' do
+      it 'rate limits login' do
+        SiteSetting.max_logins_per_ip_per_hour = 2
+        RateLimiter.stubs(:disabled?).returns(false)
+        RateLimiter.clear_all!
+
+        ActionDispatch::Request.any_instance.stubs(:remote_ip).returns('10.0.0.1')
```

Comment:
Does the `remote_ip` change in the test env? It'll be good if we can avoid stubbing here.